CMSgov/easi-app
# Easi 4116/bookmarked sys

# EASI-4116


## Changes and Description

- Add `isBookmarked` field to Cedar System
- Create dataloader flow for `isBookmarked`
- Remove duplicate `ID()` calls for EUA User ID when saving a bookmark
- Update postman collection for new `isBookmarked` field
- Also updated postman collection and stored the entire CEDAR URL in the postman variable - can undo if desired
- NOTE -- the UI is NOT updated in this PR, that will happen in a separate ticket (EASI-4138)
- NOTE 2 -- preexisting bookmark route NOT removed. It can be removed when the UI no longer uses it



## How to test this change

- Pull down branch and run tests, all should pass
- Navigate to local UI, bookmark one of our mock systems, then go to GraphQL playground (or postman, but postman is set up for USR1's EUA, so you will have to change those vars out for whatever you log in with locally - i use FAUI), and run the updated GetSystem query using the system ID that you bookmarked:
```graphql
query($systemID: String!) {
  cedarSystem(cedarSystemId: $systemID){
    id
    name
    description
    acronym
    status
    businessOwnerOrg
    businessOwnerOrgComp
    systemMaintainerOrg
    systemMaintainerOrgComp
    isBookmarked
  }
}
```
- confirm the result comes back as `isBookmarked: true`



## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
